test/kubernetes_unittest.cc

```diff
@@ -854,28 +914,288 @@ TEST_F(KubernetesTest, KubernetesUpdater) {
+    for (int i = 0; i < 3; i++) {
+      // Send a response to the watcher.
+      json::value resp = json::object({
+        {"type", json::string("ADDED")},
+        {"object", json::object({
+          {"metadata", json::object({
+            {"name", json::string("testname")},
+            {"namespace", json::string("testnamespace")},
+          })},
+          {"subsets", json::array({
+            json::object({
+              {"addresses", json::array({
+                json::object({
+                  {"targetRef", json::object({
+                    {"kind", json::string("Pod")},
+                    {"name", json::string("my-pod")},
+                  })},
+                }),
+              })},
+            }),
+          })},
+        })}
+      });
+      server.SendStreamResponse(endpoints_watch_path, resp->ToString());
+
+      // Wait until watcher has processed response (by polling the store).
+      MonitoredResource resource("k8s_cluster",
+                                 {{"cluster_name", "TestClusterName"},
+                                  {"location", "TestClusterLocation"}});
+      EXPECT_TRUE(
+          WaitForNewerCollectionTimestamp(store, resource, last_endpoints_timestamp));
+
+      // Update last_endpoints_timestamp.
+      const auto metadata_map = store.GetMetadataMap();
+      const auto& metadata = metadata_map.at(resource);
+      last_endpoints_timestamp = metadata.collected_at;
+
+      // Verify values in the store.
+      EXPECT_EQ("TestVersion", metadata.version);
+      EXPECT_FALSE(metadata.is_deleted);
+      EXPECT_EQ(Timestamp(), metadata.created_at);
+      MonitoredResource pod_mr = MonitoredResource("k8s_pod", {
+        {"cluster_name", "TestClusterName"},
+        {"namespace_name", "testnamespace"},
+        {"pod_name", "my-pod"},
+        {"location", "TestClusterLocation"},
+      });
+      json::value expected_cluster = json::object({
+        {"blobs", json::object({
+          {"services", json::array({
+            json::object({
+              {"api", json::object({
+                {"pods", json::array({
+                  pod_mr.ToJSON(),
+                })},
+                {"raw", json::object({
+                  {"metadata", json::object({
+                    {"name", json::string("testname")},
+                    {"namespace", json::string("testnamespace")},
+                  })},
+                })},
+                {"version", json::string("1.6")},  // Hard-coded in kubernetes.cc.
+              })}
+            }),
+          })},
+        })},
+      });
+      EXPECT_EQ(expected_cluster->ToString(), metadata.metadata->ToString());
+    }
   }
 
   // Terminate the hanging GETs on the server so that the updater will finish.
   server.TerminateAllStreams();
   updater_thread.join();
 }
 
+TEST_F(KubernetesTest, KubernetesUpdater) {
+  TestKubernetesUpdater(/*cluster_level_metadata=*/false);
```

Comment:
Let's split this into two separate tests.

Reply:
Done.